### PR TITLE
MOSIP-31780: Otp_manager_DB_Script changes

### PR DESCRIPTION
**Pull request by JayaSaiKrishna11**

### CODE REVIEW

**Review by LoganathanSekar7627 — APPROVED** [review]

**Comment by ckm007** [comment]
Need to add the same to upgrade scripts also...

**Comment by JayaSaiKrishna11** [comment]
@ckm007 Since this is new DB, so there's no such previous DB existing, we can't have a DB upgrade script for this.
Please suggest on this.

**Comment by ckm007** [comment]
Since it is a new DB in itself all the DB creation related statements should be part of upgrade sql for the next release version.
We also need to make the changes in the postgres upgrade part to include this new repo and add relevant changes in the upgrade.csv of postgres-init and mosip-infra.

**Review by LoganathanSekar7627 — APPROVED** [review]

File: db_upgrade_scripts/mosip_otp/sql/1.2.0.1_to _1.2.0.2_mosip_otp_upgrade.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+\COPY otp.otp_transaction FROM 'dml/otp_transaction.csv' WITH (FORMAT CSV, HEADER);
```
Comment (ckm007) [inline db_upgrade_scripts/mosip_otp/sql/1.2.0.1_to _1.2.0.2_mosip_otp_upgrade.sql:1]: this one seems to be duplicate sql file

**Comment by ckm007** [comment]
Revoke sql file is not added.

File: db_upgrade_scripts/mosip_otp/upgrade.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+CONN=$(PGPASSWORD=$SU_USER_PWD psql -v ON_ERROR_STOP=1 --username=$SU_USER --host=$DB_SERVERIP --port=$DB_PORT --dbname=$DEFAULT_DB_NAME -t -c "SELECT count(pg_terminate_backend(pg_stat_activity.pid)) FROM pg_stat_activity WHERE datname = '$MOSIP_DB_NAME' AND pid <> pg_backend_pid()";exit;)
+echo "Terminated connections"
+
+# Execute upgrade or rollback
+if [ "$ACTION" == "upgrade" ]; then
+  echo "Upgrading database from $CURRENT_VERSION to $UPGRADE_VERSION"
+  UPGRADE_SCRIPT_FILE="sql/${CURRENT_VERSION}_to_${UPGRADE_VERSION}_upgrade.sql"
+  if [ -f "$UPGRADE_SCRIPT_FILE" ]; then
+    echo "Executing upgrade script $UPGRADE_SCRIPT_FILE"
+    if [[ "$UPGRADE_VERSION" == "1.2.0.1-B1"  &&  "$CURRENT_VERSION" == "1.1.5.5" ]]; then
```
Comment (balaji-alluru) [inline db_upgrade_scripts/mosip_otp/upgrade.sh:35]: Please change the versions. current_version will be 1.2.0.1 and upgrade_version will be 1.2.0.2

File: db_upgrade_scripts/mosip_otp/sql/1.2.0.1_to _1.2.0.2_mosip_otp_upgrade.sql (hunk 1 above)
Comment (balaji-alluru) [inline db_upgrade_scripts/mosip_otp/sql/1.2.0.1_to _1.2.0.2_mosip_otp_upgrade.sql:1]: please add a rollback script, and  truncate the table in it.

**Comment by JayaSaiKrishna11** [comment]
I have created another PR and there's another ticket also regarding the same :
 https://mosip.atlassian.net/browse/MOSIP-31754. Hence closing.

**Pull request closed without merging**